operator/v1/types_vsphere_problem_detector.go

```diff
@@ -0,0 +1,71 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VSphereProblemDetector is used to manage and report health of Openshift
+// cluster running in vSphere environment.
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VSphereProblemDetector objects allows configuration and health reporting of
+// Vsphere problem detector operator.
+type VSphereProblemDetector struct {
```

Comment:
Looking at this from a bigger picture I wonder if we shouldn't think about having a generic ProblemDetector resource with specific implementations for different providers. /cc @deads2k

Reply:
This was brought up during enhancement review as well - https://github.com/openshift/enhancements/pull/532#pullrequestreview-527428457 .  

Having said that - do we have any concrete use cases for other cloudproviders? For vSphere it is plain and simple - a better monitoring and healthhecking will result in fewer bugs and since there are so many difference pieces that need to be manually configured, chance of getting something wrong is high.

Reply:
There are numerous possible problem detectors we can implement both cloud provider dependant and not. We already have some foundations within several operators of ours, see https://github.com/openshift/enhancements/pull/406, as an example. I'm pretty sure each team can expand on that and having those configurable in a central place is definitely easier for cluster admin to find and configure.